OpenSim/Common/Test/testDataTable.cpp

```diff
@@ -251,6 +268,29 @@ int main() {
             ASSERT(row2[8] == 1);
         }
 
+        {
+        std::cout << "Test feeding rows of one table to another [Vec3]."
+                  << std::endl;
+        DataTableVec3 tableVec3Copy{};
+        tableVec3Copy.setColumnLabels(tableVec3.getColumnLabels());
+        for(unsigned row = 0; row < tableVec3.getNumRows(); ++row)
+            tableVec3Copy.appendRow(tableVec3.getIndependentColumn()[row],
+                                    tableVec3.getRowAtIndex(row));
+
+        ASSERT(tableVec3Copy.getNumColumns() == tableVec3.getNumColumns());
+        ASSERT(tableVec3Copy.getNumRows()    == tableVec3.getNumRows());
+        for(unsigned r = 0; r < tableVec3.getNumRows(); ++r) {
+            for(unsigned c = 0; c < tableVec3.getNumColumns(); ++c) {
+                ASSERT(tableVec3Copy.getRowAtIndex(r).getElt(0, c)[0] ==
+                           tableVec3.getRowAtIndex(r).getElt(0, c)[0]);
+                ASSERT(tableVec3Copy.getRowAtIndex(r).getElt(0, c)[1] ==
+                           tableVec3.getRowAtIndex(r).getElt(0, c)[1]);
+                ASSERT(tableVec3Copy.getRowAtIndex(r).getElt(0, c)[2] ==
+                           tableVec3.getRowAtIndex(r).getElt(0, c)[2]);
```

Comment:
The problem here is that the following snippet throws an exception.
```c++
SimTK::Matrix_ matrix{};
/*--------------------*/
/* Fill up the matrix */
/*--------------------*/
/* Following throws   */
matrix.row(1)[0];
```
This is because `SimTK::Matrix_::row()` returns a non-const `RowVectorView` but actually represents const data. Because `RowVectorView` is non-const, the call to `operator[]` above is calling the non-const `operator[]` and throwing an exception.

I think I can fix it on `DataTable` by adding const to the `RowVectorView` that `SimTK::Matrix` returns. But it will be good to if `SimTK::Matrix` did that.